File: Leetcode/1128. Number of Equivalent Domino Pairs.cpp
class Solution {
public:
    int numEquivDominoPairs(vector<vector<int>>& dominoes) {
        map<vector<int>,int> m;
        int res=0;
        for(vector<int> i:dominoes){
            sort(i.begin(),i.end());
            m[i]++;
        }
        for(auto i=m.begin();i!=m.end();i++){
            res+=((i->second*(i->second -1))/2);
        }
        return res;
    }
};

File: Leetcode/1128. Number of Equivalent Domino Pairs.py
class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        f = {}
        res = 0
        for i in dominoes:
            n = tuple(sorted(i))
            if n in f:
                f[n] += 1
            else:
                f[n] = 1
        for i in f:
            res += (f[i] * (f[i] - 1)) // 2
        return res

        
